contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -404,7 +500,55 @@ contract MixinSignatureValidator is
             return returnData.readUint256(0) == 1;
         }
         // Static call to verifier failed.
-        _rrevert(SignatureWalletOrderValidatorError(
+        _rrevert(SignatureOrderWalletError(
+            orderHash,
+            walletAddress,
+            signature,
+            returnData
+        ));
+    }
+
+    /// @dev Verifies order AND signature via an EIP1271 Wallet contract.
+    /// @param order The order.
+    /// @param orderHash The order hash.
+    /// @param walletAddress Address that should have signed the given hash
+    ///                      and defines its own order/signature verification method.
+    /// @param signature Proof that the order has been signed by signer.
+    /// @return True if order and signature are validated by the Wallet.
+    function _validateOrderWithEIP1271Wallet(
```

Comment:
I wonder if we should consolidate this signature type with the `OrderWallet` type, since they pretty much serve the same purpose. The `OrderWallet` receiver takes an `orderHash` argument in addition to what we are doing here, which doesn't feel that necessary. That could also be encoded in the data if we want to keep it in.

Reply:
I am down for this, now that I see that decoding is nbd.